project-koku/koku
# Enqueue separate celery task for each month of data

[issue_1478_ut.txt](https://github.com/project-koku/koku/files/3946555/issue_1478_ut.txt)
Addresses #1478 

Recent findings with light house customers in production have proven that processing multiple months of data in a single celery task is inefficient.

This change enqueues a separate celery download/process task for each month of data.

**Testing**
1.  Create an AWS source and verify that the current and previous months download/process flow is kicked off in seperate celery tasks.
2.  Add Azure provider and verify processing is successful.
3.  Add an OCP provider and generate and push data throguh ingress service.  Verify it's successful


**Testing Results**
[issue_1478_ut.txt](https://github.com/project-koku/koku/files/3946557/issue_1478_ut.txt)



Vortex API tests passed
Running IQE smoke tests now...
